kore/src/Kore/OnePath/Verification.hs

```diff
@@ -77,6 +77,38 @@ import qualified Kore.TopBottom as TopBottom
 import           Numeric.Natural
                  ( Natural )
 
+{- NOTE: Non-deterministic semantics
+
+The current implementation of one-path verification assumes that the definition
+semantics is deterministic, that is: at most one rule may apply to any
```

Comment:
Well, it's a bit more complicated, non-determinism is allowed as long as you have confluence and your final state not in the middle of a non-confluent thing. As an example, you can have non-deterministic rewriting for heating and cooling, but usually you can reach the proof goal only after the non-deterministic part ended, so all is fine.

It's up to you whether to rewrite this in a more precise way.

Reply:
I think this is an important distinction that I did not fully understand, so I will try to be more precise.